### DISCUSSION TITLE
Unable to get strongSwan 5.9.5 as road warrior to work with Cisco ASA 9.19(1)28

### DISCUSSION THREAD

**Discussion by girhub0** [strongswan/strongswan#2393] [General]
Hi.

First of all, I have tried setting up strongSwan and the Cisco ASA to use PSK for strongSwan and certificate for the Cisco ASA.
That works, so everything related to certificate, proposals etc. is good.

Next, I want to get username/password to work, instead of PSK. Client certificate may be tested later.

ASA is configured like this for this part of the test:

> tunnel-group DefaultRAGroup ipsec-attributes
>  chain
>  ikev2 rsa-sig-hash sha1 
>  ikev2 remote-authentication eap query-identity
>  ikev2 local-authentication certificate ASDM_TrustPoint
> tunnel-group DefaultRAGroup ppp-attributes
>  no authentication chap
>  no authentication ms-chap-v1
>  authentication ms-chap-v2

 
 
 strongSwan is configured like this:
 
 

> connections {
>   home {
>    version=2
>    remote_addrs = vpn.test.com
>    vips=0.0.0.0
>    local {
>     auth = eap
>     eap_id = test
>     id = test
>    }
>    remote {
>     auth = pubkey
>     id = "vpn.test.com"
>    }
>    children {
>     home {
>      remote_ts  = 0.0.0.0/0
>      start_action = start
>     }
>    }
>   }
> }
> 
> secrets {
>   eap-test {
>    id = test
>    secret = secret
>   }
> }

Syslog from strongSwan says:

> server requested EAP_IDENTITY (id 0x24), sending 'test'
> generating IKE_AUTH request 2 [ EAP/RES/ID]
> received AUTHENTICATION_FAILED notify error

ASA syslog says:

> Username:test IKEv2 Negotiation aborted due to ERROR: Authenticator sent NULL EAP message

I have tried to change "auth" from "eap" to "eap-mschapv2" in strongSwan, but the same happens.

Any tips on how to resolve this?

**Comment by tobiasbrunner** [comment]
Sounds like there might be a problem with your RADIUS server (maybe it doesn't like the EAP identity, `test`, that's being sent). Check the log there.

**Reply by girhub0** [reply]
I have tested with RADIUS previously, and also got "NULL EAP". To simplify the test, I have removed RADIUS and I'm using ASA's local user database for authentication.

The username of the test user is not "test", that's just what I posted here. I have tried different usernames, but ASA still says that strongSwan sends NULL EAP.

**Reply by tobiasbrunner** [reply]
No idea, sorry. You have to ask Cisco about that.

**Reply by girhub0** [reply]
Will do.

I'm a bit pessimistic though, because as of yet, I have been unable to find any guides anywhere for how to get strongSwan to work as an IKEv2 client towards Cisco equipment.

Neither Cisco nor strongSwan mentions the possibility anywhere, and I haven't found anything in this or other fora.
